library/java/src/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -50,7 +56,9 @@ String resolveTemplate(String templateYAML) {
             .replace("{{ dns_failure_refresh_rate_seconds_max }}",
                      String.format("%s", dnsFailureRefreshSecondsMax))
             .replace("{{ stats_flush_interval_seconds }}", String.format("%s", statsFlushSeconds))
-            .replace("{{ device_os }}", "Android");
+            .replace("{{ device_os }}", "Android")
+            .replace("{{ app_version }}", String.format("%s", appVersion))
+            .replace("{{ app_id }}", String.format("%s", appId));
```

Comment:
Shouldn't this work?
```suggestion
            .replace("{{ app_version }}", appVersion)
            .replace("{{ app_id }}", appId);
```